### PR TITLE
Unit tests for token validation

### PR DESCRIPTION
**Pull request by petertgiles**
This branch adds two sets of unit tests to validate the token validation logic for both local auth and Sign In Canada.
To run the tests, navigate to the API folder and run:
`./vendor/bin/phpunit tests/OpenIdBearerTokenTest.php`
`./vendor/bin/phpunit tests/LocalAuthBearerTokenTest.php`
All the tests in these two classes should be successful right now.
To make this possible, some refactoring has been applied to the bearer token processing.

Two keys with corresponding certs have been created for running tests and generating keys.  Tokens for testing have been generated using [jwt.io](https://jwt.io/).

### CODE REVIEW

**Comment by brindasasi** [comment]
Not much into php.  But wondering hard coding the tokens that almost never expire ( looks like 16 years) would  impose any security concern ?

**Comment by petertgiles** [comment]
> wondering hard coding the tokens that almost never expire ( looks like 16 years) would impose any security concern ?

Thanks for taking a look!  You're right that these keys and the tokens almost never expire though that's intentional so that the tests behave reliably into the future.  These keys and tokens are only used in the tests and not in the regular operation of the site so I think there's no security risk.  Let me know if you have some ideas though. 😃

File: api/tests/LocalAuthBearerTokenTest.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        parent::setUp();
+
+    }
+    /**
+     * @test
+     * A valid token is provided and validated.  The test checks that the right sub value is returned.
+     */
+    public function it_accepts_a_good_token_and_returns_the_right_sub()
+    {
+        //$token = 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwOi8vdGVzdC5jb20iLCJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjowLCJleHAiOjIxNDc0ODM2NDd9.G8HuIKwidOxRTgTKQCw1Lo_rU0j8PD57i9TfY5RIYqo3osCILwyDKCd31f6luZvy49PT4R_x-_ZOFMmkb6_pnuJGSHYv_EpZcYWCrI2X5USox4IrQ4mmFGZP8pPJOeBIAS00i_Unab-LR2mnqNCT-EsZOA7sHcJTDyNB4FSsHtmZNKXtsw335DSuDb6nIa77SV1gmhPIA7LKnxgSvacxtSDr9LbEqX-mwXfMSxTOWsLosBwqHHjc2W3tWlPHEAkLqGMsEl6miZK7RSAPXUE16vlGM9QccCidZWXBXi9lZ_tVkHIIsRIXdiS1RTc8VMtumi1_vRMFU4dheypRQQ68ww';
```
Comment (mnigh) [inline api/tests/LocalAuthBearerTokenTest.php:34]: Can this be removed?

File: api/tests/LocalAuthBearerTokenTest.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        $this->service_provider = new LocalAuthBearerTokenService($issuer, $publicKey, $timeZone);
+
+        parent::setUp();
+
+    }
+    /**
+     * @test
+     * A valid token is provided and validated.  The test checks that the right sub value is returned.
+     */
+    public function it_accepts_a_good_token_and_returns_the_right_sub()
```
Comment (mnigh) [inline api/tests/LocalAuthBearerTokenTest.php:32]: Our naming convention for PHP functions is to use camelCase (for example, [testCreateUserDefaultRoles](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/tests/UserTest.php#L12)). I don't think this is documented anywhere, is it @tristan-orourke?

**Review by mnigh — COMMENTED** [review]
A few syntax comments to address but otherwise, looks good to me.

File: api/tests/LocalAuthBearerTokenTest.php (hunk 1 above)
Comment (petertgiles) [reply api/tests/LocalAuthBearerTokenTest.php]: Good catch.  This was an old token that I forgot to remove.

File: api/tests/LocalAuthBearerTokenTest.php (hunk 2 above)
Comment (petertgiles) [reply api/tests/LocalAuthBearerTokenTest.php]: Yeah, you're right.  All the rest of the code uses camelCase.  I've updated it.

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**